Apps/Common/src/Services/NotificationSettingsService.cs

```diff
@@ -118,8 +119,16 @@ private static NotificationSettingsRequest ValidateSMSVerificationCode(Notificat
             if (notificationSettings.SMSEnabled && string.IsNullOrEmpty(notificationSettings.SMSVerificationCode))
             {
                 // Create the SMS validation code if the SMS is not verified and the caller didn't set it.
-                Random generator = new Random();
-                notificationSettings.SMSVerificationCode = generator.Next(0, 999999).ToString("D6", CultureInfo.InvariantCulture);
+                using (RandomNumberGenerator generator = RandomNumberGenerator.Create())
+                {
+                    byte[] data = new byte[16];
```

Comment:
This array only needs to have a length of 4, the size of a UInt32.

Reply:
+1